### PR TITLE
Feature/sar 385 time picker

### PR DESCRIPTION
**Pull request by eawest2**
# Related Tickets

- [SAR-385](https://amida.atlassian.net/browse/SAR-385)

# How Things Worked (or Didn't) Before This PR

You couldn't filter display results by date.

# How Things Work Now (And How to Test)

Now you can. 

Click on the timeline button in the ChartBar, and select a date range in which some data is presented but not all. Then look at how the graph changes. 

Click the clear button to reset it back to standard.

# Considerations

The use of the MUI date-range picker required the integration of `moment.js` or another time-function package from a list on their page. This may not be necessary if we are opposed to custom ranges of view, but is if we intend to use the boxed MUI components.

MUI's lab components aren't quite fully wired for styling just yet, meaning some janky manual styling had to be done to the chartbar, which included the `!important` flag. This is not best practice, and will need to be addressed later, but Keith and I spent over an hour digging around and didn't find another option that actually worked.

As this issue is a major MVP feature-add, I'm putting it up for review with the knowledge that it could use some improvement, I'm documenting them here for posterity.

# Readiness

1. [x] This PR passes all automated tests.
2. [x] This PR has no linting errors.
3. [x] This PR's changes to configuration files have been documented in all appropriate places (such as but not limited to README.md), if applicable.

### CODE REVIEW

File: src/components/ChartContainer/D3Container.js [hunk 1]
```diff
@@ -78,6 +80,23 @@ function D3Container({ dashboardState, dashboardActions, store }) {
         )),
       )
     }
+    if (dates.startDate || dates.endDate) {
```
Comment (anthem123) [inline src/components/ChartContainer/D3Container.js:83]: I'm getting an issue when using the other filters that `dates` is undefined.

`if (dates && (dates.startDate || dates.endDate))` only half fixes the problem though. After setting a date range other filters the date range keeps reseting.

Filtering by only dates though, perfect.

**Comment by timAmida** [comment]
Looks Good to Me

**Review by timAmida — APPROVED** [review]
Looks good

**Review by anthem123 — APPROVED** [review]
Date filter now works as expected. Approved.

**Pull request merged**